java/tools/src/main/java/org/apache/arrow/tools/FileToStream.java

```diff
@@ -44,12 +44,11 @@ public static void convert(FileInputStream in, OutputStream out) throws IOExcept
     try (ArrowFileReader reader = new ArrowFileReader(in.getChannel(), allocator)) {
       VectorSchemaRoot root = reader.getVectorSchemaRoot();
       // load the first batch before instantiating the writer so that we have any dictionaries
-      if (!reader.loadNextBatch()) {
-        throw new IOException("Unable to read first record batch");
-      }
+      // only writeBatches if we loaded one in the first palce.
+      boolean writeBatches = reader.loadNextBatch();
```

Comment:
is this not different from the behavior earlier - we would have thrown exception earlier for zero batches vs what it looks like proceeding silently now?

Reply:
Yes, this is intentional.  This was discussed on the ML, we think it is reasonable to have a schema without batches.

Reply:
would it not be better to do that in a different method? I am not sure if existing clients depend on this for validation?

Reply:
Maybe.  Given the fact that there are no unit tests for this code path makes me think this this is only used in integration tests. Are you familiar with any additional use cases?  Also this change seems like the correct behavior (I don't see why no batches is an exception) but I suppose it would be better to validate at the file level that there wasn't any corruption (but I don't think there are provisions for this in the spec)

Reply:
Not sure of exact use cases, but have been dealing with a lot of breaking changes while upgrading dremio to use latest arrow and am wary of anything that looks like change in behavior :)  

Agree the new behavior seems like the right one to implement. Sounds good for now.

Reply:
Understood at some point we should discuss on the ML about API/behavior stability